File: frontend/types/idea.ts
export interface IdeaRequest {
    title: string;
    industry: string;
    targetAudience: string;
    description: string
}

export interface IdeaResponse {
    startup_name: string;
    elevator_pitch: string;
    problem_statement: string;
    solution: string;
    target_audience: string;
    business_model: string;
    mvp_features: string[];
}

File: backend/app/schemas/idea.py
from pydantic import BaseModel, Field

class IdeaRequest(BaseModel):
    title: str= Field(...,min_length=5,max_length=100)
    industry: str= Field(...,min_length=2,max_length=50)
    targetAudience: str= Field(...,min_length=3,max_length=100)
    description: str= Field(...,min_length=30)

class IdeaResponse(BaseModel):
    startup_name: str
    elevator_pitch: str
    problem_statement: str
    solution: str
    target_audience: str
    business_model: str
    mvp_features: list[str]